### PR TITLE
{tools} [foss/2018a] ipyparallel v6.2.2 for Python/3.6.4

### PR DESCRIPTION
**Pull request by SimonPinches**

### CODE REVIEW

File: easybuild/easyconfigs/i/ipyparallel/ipyparallel-6.2.2-foss-2018a-Python-3.6.4.eb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    ('Python', '3.6.4'),
+    ('IPython', '6.4.0', versionsuffix),
+]
+
+# raise an error when auto-downloading of dependency packages is detected
+exts_download_dep_fail = True
+
+exts_defaultclass = 'PythonPackage'
+exts_list = [
+    (name, version, {
```
Comment (boegel) [inline easybuild/easyconfigs/i/ipyparallel/ipyparallel-6.2.2-foss-2018a-Python-3.6.4.eb]: @SimonPinches If only one single Python package is involved, there's no need to use `Bundle`, just use `PythonPackage` directly (which significantly simpliies the easyconfig).

We should get this properly documented...

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/i/ipyparallel/ipyparallel-6.2.2-foss-2018a-Python-3.6.4.eb [hunk 2]
```diff
@@ -0,0 +1,39 @@
+easyblock = 'PythonPackage'
+
+name = 'ipyparallel'
+version = '6.2.2'
+python = "Python"
+pythonversion = "3.6.4"
+versionsuffix = "-%s-%s" % (python, pythonversion)
```
Comment (boegel) [inline easybuild/easyconfigs/i/ipyparallel/ipyparallel-6.2.2-foss-2018a-Python-3.6.4.eb]: @SimonPinches Please use `%(pyver)s`:

```
version = '6.2.2'
versionsuffix = '-Python-%(pyver)s'
...
dependencies = [
    ('Python', '3.6.4'),
...
```

File: easybuild/easyconfigs/i/ipyparallel/ipyparallel-6.2.2-foss-2018a-Python-3.6.4.eb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+full_sanity_check = True
+
+modextrapaths = {'PYTHONPATH': ['lib/python%(pyshortver)s/site-packages']}
+
+sanity_check_paths = {
+    'files': [],
+    'dirs': ['lib/python%(pyshortver)s/site-packages'],
+}
+
+sanity_check_commands = ["python -c 'import ipyparallel'"]
```
Comment (boegel) [inline easybuild/easyconfigs/i/ipyparallel/ipyparallel-6.2.2-foss-2018a-Python-3.6.4.eb]: @SimonPinches No need for this, this is done automatically by `PythonPackage` (the check is basically done twice now, no need for that).

File: easybuild/easyconfigs/i/ipyparallel/ipyparallel-6.2.2-foss-2018a-Python-3.6.4.eb [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    ('IPython', '6.4.0', versionsuffix),
+]
+
+use_pip = True
+
+exts_download_dep_fail = True
+
+full_sanity_check = True
+
+modextrapaths = {'PYTHONPATH': ['lib/python%(pyshortver)s/site-packages']}
```
Comment (boegel) [inline easybuild/easyconfigs/i/ipyparallel/ipyparallel-6.2.2-foss-2018a-Python-3.6.4.eb]: @SimonPinches No need for this, this is done automatically by `PythonPackage`

File: easybuild/easyconfigs/i/ipyparallel/ipyparallel-6.2.2-foss-2018a-Python-3.6.4.eb [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+dependencies = [
+    (python, pythonversion),
+    ('IPython', '6.4.0', versionsuffix),
+]
+
+use_pip = True
+
+exts_download_dep_fail = True
+
+full_sanity_check = True
```
Comment (boegel) [inline easybuild/easyconfigs/i/ipyparallel/ipyparallel-6.2.2-foss-2018a-Python-3.6.4.eb]: @SimonPinches This is specific to `Bundle`, it's meaningless here

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by SimonPinches** [comment]
Hi @boegel, I've cleaned up the `eb` file for `ipyparallel` as you indicated.  Hopefully soon I'll start producing cleaner files in the first place!

File: easybuild/easyconfigs/i/ipyparallel/ipyparallel-6.2.2-foss-2018a-Python-3.6.4.eb [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+checksums = ['02b225966d5c20f12b1fba0b6b10aa5d352a6b492e075f137ff0ff6e95b9358e']
+
+dependencies = [
+    ('Python', "3.6.4"),
+    ('IPython', '6.4.0', versionsuffix),
+]
+
+use_pip = True
+
+exts_download_dep_fail = True
```
Comment (boegel) [inline easybuild/easyconfigs/i/ipyparallel/ipyparallel-6.2.2-foss-2018a-Python-3.6.4.eb]: @SimonPinches This should be `download_dep_fail`, no extensions involved here...

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/i/ipyparallel/ipyparallel-6.2.2-foss-2018a-Python-3.6.4.eb (hunk 6 above)
Comment (SimonPinches) [reply easybuild/easyconfigs/i/ipyparallel/ipyparallel-6.2.2-foss-2018a-Python-3.6.4.eb]: @boegel, fixed :-)

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2606.swalot.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/036a85f6a63b008a6e6370114f4e2376 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2001.delcatty.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/43cc95444716414a32ec88c0a79b5b66 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node3112.skitty.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/f4083c09b5117dec04f8f6b76474bfc4 for a full test report.

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
Going in, thanks @SimonPinches!

**Pull request merged**